src/database/index.tsx

```diff
@@ -482,3 +486,75 @@ export const deleteFacility = async (
     console.error(error);
   }
 };
+
+export const duplicateScenario = async (
+  scenarioId: string,
+): Promise<Scenario | void> => {
+  try {
+    const scenario = await getScenario(scenarioId);
+
+    if (!scenario) {
+      console.error(`No scenario found for scenario: ${scenarioId}`);
+      return;
+    }
+
+    const userId = currrentUserId();
+    const timestamp = currrentTimestamp();
+    const db = await getDb();
+    const batch = db.batch();
+
+    // Duplicate and save the Scenario
+    const scenarioDoc = db.collection(scenariosCollectionId).doc();
+
+    batch.set(scenarioDoc, {
+      name: `Copy of ${scenario.name}`,
+      description: `This is a copy of the '${
+        scenario.name
+      }' scenario, made on ${format(new Date(), MMMMdyyyy)}`,
+      baseline: false,
+      dailyReports: false,
+      dataSharing: false,
+      promoStatuses: {},
+      roles: {
+        [userId]: "owner",
+      },
+      createdAt: timestamp,
+      updatedAt: timestamp,
+    });
+
+    // Duplicate and save all of the Facilities
+    const facilities = await getFacilities(scenarioId);
+    for (const facility of facilities) {
```

Comment:
To be clear, we're _not_ overwriting the `updatedAt` or `observedAt` timestamps on `modelVersions` because we want to preserve that chain of observations. Makes sense. Is the reason we are not updating the `facility.updatedAt` timestamp just to make sure we don't lose its history? Makes sense to me but want to be sure.

Reply:
Correct, we are not overwriting the `updatedAt` or `observedAt` in `modelVersions` so that we can preserve the observation history.  This is significant so that we don't mess up R(t) calculations in these copies.  There isn't a strict use case for preserving the `facility.createdAt` and `facility.updatedAt` per se, but I chose to preserve them because I thought it would look a bit weird if the duplicated facilities all looked like they were updated at the time of duplication in the Facility Row instead of at the actual time they were last updated (see the image below):

<img width="819" alt="Screen Shot 2020-05-01 at 4 24 18 PM" src="https://user-images.githubusercontent.com/25503/80839026-5d1b1f00-8bc8-11ea-998f-f9f012148a5d.png">

If that was an errant judgment call, I'm happy to drop these timestamps and generate new ones for the facility copies.